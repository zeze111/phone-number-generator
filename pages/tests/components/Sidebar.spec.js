import React from "react";
import { shallow } from "enzyme";
import Sidebar from "../../components/sidebar";

describe("Sidebar", () => {
  it("should render the component", () => {
    const props = {
      linkNo: 1,
      setLinkNo: jest.fn()
    };

    const wrapper = shallow(<Sidebar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe("setLinkNo", () => {
    it("should set the state of linkNo", () => {
      const props = {
        linkNo: 1,
        setLinkNo: jest.fn()
      };
      const wrapper = shallow(<Sidebar  {...props} />);
      const link = 2;
      const button = wrapper.find("#list");
      button.simulate("click");
      expect(props.setLinkNo).toHaveBeenCalledWith(link);
    });
  });
});
