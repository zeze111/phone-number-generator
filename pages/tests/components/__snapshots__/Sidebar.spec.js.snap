// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render the component 1`] = `
<div
  className="sidebar-div"
>
  <button
    className="pressed"
    id="generate"
    onClick={[Function]}
  >
    <p
      className="sidebar-inner-text2"
    >
      Generate Numbers
    </p>
  </button>
  <button
    className="sidebar-list"
    id="list"
    onClick={[Function]}
  >
    <p
      className="sidebar-inner-text2"
    >
      Numbers
    </p>
  </button>
  <button
    className="sidebar-list"
  >
    <p
      className="sidebar-inner-text2"
    >
      Accounts
    </p>
  </button>
  <button
    className="sidebar-list"
  >
    <p
      className="sidebar-inner-text2"
    >
      Settings
    </p>
  </button>
</div>
`;
